src/globus_sdk/services/tunnels/client.py

```diff
@@ -0,0 +1,330 @@
+        if restartable is not None:
+            attributes["restartable"] = restartable
+        if lifetime_mins is not None:
+            attributes["lifetime_mins"] = lifetime_mins
+
+        data = {
+            "data": {
+                "type": "Tunnel",
+                "attributes": attributes,
+                "relationships": {
+                    "listener": {
+                        "data": {
+                            "type": "StreamAccessPoint",
+                            "id": listener_stream_ap_id,
+                        }
+                    },
+                    "initiator": {
+                        "data": {
+                            "type": "StreamAccessPoint",
+                            "id": initiator_stream_ap_id,
+                        }
+                    },
+                },
+            }
+        }
+
+        r = self.post("/v2/tunnels", data=data, query_params=query_params)
+        return r
+
+    def show_tunnel(
+        self,
+        tunnel_id: str,
+        *,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param tunnel_id: The ID of the Tunnel which we are fetching details about.
+        :param query_params: Any additional parameters will be passed through
+            as query params.
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    result = tc.show_tunnel(tunnel_id)
+                    print(result["data"])
+
+            .. tab-item:: API Info
+
+                ``GET /tunnels/<tunnel id>``
+        """
+        r = self.get(f"/v2/tunnels/{tunnel_id}", query_params=query_params)
+        return r
+
+    def set_listener(
+        self,
+        tunnel_id: str,
+        listener_cs: str,
+        *,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param tunnel_id: The ID of the Tunnel which the application listener's
+            contact string will be set.
+        :param listener_cs: The contact string of the application listener's
+            TCP socket in the form <host>:<port>.
+        :param query_params: Any additional parameters will be passed through
+            as query params.
+
+        This tells the Tunnel where the application is listening for connections
+        through the tunnel. Once this is set the Tunnel can move to the ACTIVE
+        state.
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    tc.set_listener(tunnel_id, "192.168.0.23:8080")
+
+            .. tab-item:: API Info
+
+                ``PATCH /tunnels/<tunnel id>``
+        """
+        cs = listener_cs.split(":")
+        host = cs[0]
+        port = cs[1]
```

Comment:
I recommend this form to avoid `KeyError` in the case that someone leaves out the port, or perhaps uses IPv6 (if that's supported). After that, it may be that the host needs to be stripped of square brackets (though I'm not an expert on IPv6; I simply saw in a cursory search that IPv6 addresses with ports have square brackets around the host, like `[2008:2007::1]:8080`).

```suggestion
        host, _, port = listener_cs.rpartition(":")
```